colin/core/checks/labels.py

```diff
@@ -73,3 +75,43 @@ def check(self, target):
                            reference_url=self.reference_url,
                            check_name=self.name,
                            logs=[])
+
+
+class InheritedOptionalLabelAbstractCheck (ImageAbstractCheck):
+
+    def __init__(self, message, description, reference_url, tags):
+        """
+        Abstract check for Dockerfile/Image labels.
+
+        :param message: str
+        :param description: str
+        :param reference_url: str
+        :param tags: [str]
+        """
+        super(InheritedOptionalLabelAbstractCheck, self) \
+            .__init__(message, description, reference_url, tags)
+
+    def check(self, target):
+        optional_labels = [
+            'summary',
+            'description',
+            'io.k8s.description (same as description)',
+            'io.k8s.display-name',
+            'io.openshift.tags',
+        ]
+        passed = True
+
+        if target.parent_target:
+            for label in optional_labels:
+                if label in target.labels:
+                    if label in target.parent_target.labels:
+                        if target.labels[label] == target.parent_target.labels[label]:
```

Comment:
It took me a while to see what's going on. What about using set operations to make it clear?

```python
        if target.parent_target:
            labels_to_check = set(optional_labels) - set(target_labels) - set(target.parent_target.labels)
            for label in labels_to_check:
                if target.labels[label] == target.parent_target.labels[label]:
```

Reply:
sure I can use set operation

but we will need intersection not difference

Reply:
Yes, you're right...;-)